quickwit/quickwit-control-plane/src/grpc_adapter.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use async_trait::async_trait;
+use quickwit_actors::Mailbox;
+use quickwit_proto::control_plane_api::control_plane_service_server::{self as grpc};
+use quickwit_proto::control_plane_api::{NotifyIndexChangeRequest, NotifyIndexChangeResponse};
+use quickwit_proto::tonic;
+
+use crate::scheduler::IndexingScheduler;
+
+#[allow(missing_docs)]
+#[derive(Clone)]
+pub struct GrpcControlPlaneAdapter(Mailbox<IndexingScheduler>);
```

Comment:
It is starting to be a bit lame that our actor framework is not distributed.

Reply:
Indeed. Maybe a task for @ChillFish8 :)